src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/DeleteCompleteFilesTask.java

```diff
@@ -80,58 +85,63 @@ private void processCompleteFiles(CloudBlobContainer container) {
 
         log.info(
             "Finished deleting complete files in container {}, deleted {} files, failed to delete {} files",
-            container.getName(),
+            container.getBlobContainerName(),
             successCount,
             failureCount
         );
     }
 
-    private boolean tryProcessCompleteEnvelope(CloudBlobContainer container, Envelope envelope) {
-        boolean deleted = false;
+    private boolean tryProcessCompleteEnvelope(BlobContainerClient container, Envelope envelope) {
 
-        String loggingContext = "File name: " + envelope.getZipFileName() + ", Container: " + container.getName();
+        String loggingContext = "File name: " + envelope.getZipFileName()
+            + ", Container: " + container.getBlobContainerName();
 
         try {
             log.info("Deleting file. {}", loggingContext);
 
-            CloudBlockBlob cloudBlockBlob = container.getBlockBlobReference(envelope.getZipFileName());
+            BlobClient blobClient = container.getBlobClient(envelope.getZipFileName());
 
-            if (cloudBlockBlob.exists()) {
-                Optional<String> leaseId = blobManager.acquireLease(
-                    cloudBlockBlob,
-                    container.getName(),
-                    envelope.getZipFileName()
+            if (Boolean.TRUE.equals(blobClient.exists())) {
+
+                leaseAcquirer.ifAcquiredOrElse(
+                    blobClient,
+                    leaseId -> delete(blobClient, leaseId),
+                    DeleteCompleteFilesTask::throwBlobDeleteException,
+                    false
                 );
-                if (leaseId.isPresent()) {
-                    AccessCondition accessCondition = generateLeaseCondition(leaseId.get());
-                    deleted = cloudBlockBlob.deleteIfExists(
-                        NONE,
-                        accessCondition,
-                        null,
-                        null
-                    );
-                    if (deleted) {
-                        log.info("File deleted. {}", loggingContext);
-                    } else {
-                        log.info("File has not been deleted. {}", loggingContext);
-                    }
-                }
+
             } else {
-                deleted = true;
                 log.info("File has already been deleted. {}", loggingContext);
             }
 
-            if (deleted) {
-                envelope.setZipDeleted(true);
-                envelopeRepository.saveAndFlush(envelope);
-                log.info("Marked envelope as deleted. {}", loggingContext);
-            }
+            envelope.setZipDeleted(true);
+            envelopeRepository.saveAndFlush(envelope);
+            log.info("Marked envelope as deleted. {}", loggingContext);
 
-            return deleted;
+            return true;
 
         } catch (Exception ex) {
```

Comment:
The logic looks simpler than it used to be regarding returning _true_ or _false_ and it makes sense to catch exception (and log it) in the calling method rather than check if the result is _true_ or _false_.